drivers/hmis_external_apis/app/models/hmis_external_apis/external_forms/form_submission.rb

```diff
@@ -80,29 +80,29 @@ def run_form_processor(current_user, project: nil)
         household_id = form_values['Enrollment.householdId']
         relationship_to_hoh = form_values['Enrollment.relationshipToHoH']
 
-        # drop household id if it doesn't match expected format
+        # household id doesn't match expected format
         hh_invalid_format = household_id && !(household_id.starts_with?('HH') && household_id.size > 20)
-        # drop household id if it has a conflict in this data source in another project (just to be safe)
-        hh_invalid_exists = household_id && Hmis::Hud::Enrollment.where(household_id: household_id, data_source: project.data_source).where.not(project: project).exists?
+        # household id has a conflict in this data source in another project (just to be safe)
+        hh_conflict = household_id && Hmis::Hud::Enrollment.where(household_id: household_id, data_source: project.data_source).where.not(project: project).exists?
 
-        if hh_invalid_format || hh_invalid_exists
+        if hh_invalid_format || hh_conflict
           household_id = nil
           relationship_to_hoh = nil
-          values_to_process.delete('Enrollment.householdId')
-          values_to_process.delete('Enrollment.relationshipToHoH')
         end
 
-        if relationship_to_hoh && !Types::HmisSchema::Enums::Hud::RelationshipToHoH.values.keys.include?(relationship_to_hoh)
-          relationship_to_hoh = nil
-          values_to_process.delete('Enrollment.relationshipToHoH')
-        end
+        # Relationship to HoH is expected to be enum format, eg 'SELF_HEAD_OF_HOUSEHOLD'. This translates it to integer value (SELF_HEAD_OF_HOUSEHOLD => 1), or nil if invalid
```

Comment:
Ah, yeah I ran up against this too and I like your fix better. After you merge this I'll revert the changes on my pr. https://github.com/greenriver/hmis-warehouse/pull/4680/files#diff-34c13c8335974559751928b36cbca0a3363274e0103ed5062da19bf038a918a7R93-R106

Reply:
Okay. I started with changing it to integers, too, but I realized that would quickly become incompatible if we had a form that _did_ collect Relationship To HoH. So I think expecting the enums may be more future-compatible (though it's kind of ugly). Sounds good